### ISSUE TITLE
Replace uses of @ivar in interfaces with z.i.Attribute

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1852]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)| @jyknight reported|
|-|-|
|Trac ID|trac#1852|
|Type|defect|
|Created|2006-06-24 02:40:37Z|
|Branch|https://github.com/twisted/twisted/tree/ivar-interfaces-1852-2|

yep, that.

Attachments:

* [1852.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/4ff/4ffba05e292341e54f6b149967cf0e728d4f11e8/e110f10e06b486b4e744a8063c09854382dda2f5.diff) (551 bytes) - added by therve on 2008-09-06 08:49:41Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__1852 1852
type__defect defect
reporter__jknight jknight
priority__lowest lowest
milestone__ 
branch__branches_ivar_interfaces_1852_2 branches/ivar-interfaces-1852-2
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1151116837000000 1151116837000000
changetime__1222276042000000 1222276042000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph commented|
|-|-|

This should probably be done eventually, but I can't imagine that it's on my list of top 1000 things to do for Twisted :).

Anyone else should feel free to steal.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

Can someone give me an example of when and where to replace what so I can work on this?

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

The attached patch should give you an example of what to do. Feel free to get back to me if it's not clear.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

Replying to [therve](#note_3):
> The attached patch should give you an example of what to do. Feel free to get back to me if it's not clear.

Thanks therve, that's clear now.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

(In [24700]) Branching to 'ivar-interfaces-1852'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

Convention thus far has been to include the `@type` information in the `Attribute` docstring.  It seems pydoctor ignores the type information for an `Attribute`, anyway.

`twisted.web2.iweb.IChanRequest.getHostInfo` isn't valid for pydoctor.  epytext should allow `C{}` to span multiple lines, but for some reason this generates an error from pydoctor.

`IGroup.join` doesn't need `pass` anymore.  Some methods in `twisted/web2/iweb.py` also have redundant `pass` statements.

Whitespace change before the `id` method in `twisted/trial/itrial.py` should be reverted

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Replying to [exarkun](#note_7):
> Convention thus far has been to include the `@type` information in the `Attribute` docstring.  It seems pydoctor ignores the type information for an `Attribute`, anyway.

Not sure what this means, left it as it is, let me know if you want it included or..

> `twisted.web2.iweb.IChanRequest.getHostInfo` isn't valid for pydoctor.  epytext should allow `C{}` to span multiple lines, but for some reason this generates an error from pydoctor.

Removed the C{} markup again.

>
> `IGroup.join` doesn't need `pass` anymore.  

removed that ```pass``` in twisted/words/im/interfaces.py

>Some methods in `twisted/web2/iweb.py` also have redundant `pass` statements.
>
Removed the redundant statements, except for the ones in ```IOldRequest``` since they don't have any docstrings.

> Whitespace change before the `id` method in `twisted/trial/itrial.py` should be reverted

Fixed, see r24836

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra commented|
|-|-|

Replying to [thijs](#note_9):
> Removed the redundant statements, except for the ones in ```IOldRequest``` since they don't have any docstrings.

See [#3416](https://github.com/twisted/twisted/issues/3416), where exarkun will probably add those missing docstrings.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

Replying to [thijs](#note_9):
> Replying to [exarkun](#note_7):
> > Convention thus far has been to include the `@type` information in the `Attribute` docstring.  It seems pydoctor ignores the type information for an `Attribute`, anyway.
>
> Not sure what this means, left it as it is, let me know if you want it included or..
>

It means don't do this:

```python
class IFoo(Interface):
    """
    blah blah 

    @type bar: C{int}
    """
    bar = Attribute("number of bars")
```

Instead, do this:

```python

class IFoo(Interface):
    """
    blah blah
    """
    bar = Attribute("An C{int} giving the number of bars")
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun commented|
|-|-|

Also, there's a conflict with trunk now, please merge forward.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_13"></a>|@thijstriemstra commented|
|-|-|

(In [24867]) Branching to 'ivar-interfaces-1852-2'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_14"></a>|@thijstriemstra removed owner|
|-|-|

Included the type information in the Attributes for [twisted/words/im/interfaces.py](twisted/words/im/interfaces.py), see r24869

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_15"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

  * twisted/words/im/interfaces.py
    * The `I{}` which was in the `IGroup` docstring and is now the docstring for that interface's `account` attribute doesn't really make sense.  It should be `I{account}` or `L{account<IAccount>}`.  This is repeated in the docstring for `IGroup.__init__`.
    * The docstring for `IGroup.sendGroupMessage` used to have two levels of indentation for _style_ and _emote_ (emote is hierarchically inferior to style); now it has one, this should go back to how it was.
  * twisted/internet/interfaces.py
    * `IReactorCore` has `L{Reactor}` in its docstring now, but there is nothing named `Reactor` that can be linked to.
  * twisted/spread/interfaces.py - changes here don't seem significant

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_16"></a>|@thijstriemstra removed owner|
|-|-|

Made the changes in r24881

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_17"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

Looks fine, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_18"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [24916]) Merge ivar-interfaces-1852-2: Replace uses of @ivar in interfaces with z.i.Attribute

Author: thijs
Reviewer: therve, exarkun
Fixes: [#1852](https://github.com/twisted/twisted/issues/1852)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
